client/src/Backend/GameLogic/GameManager.ts

```diff
@@ -4924,33 +4926,36 @@ class GameManager extends EventEmitter {
     const fromActiveArtifact = this.getActiveArtifact(fromPlanet);
     const toActiveArtifact = this.getActiveArtifact(toPlanet);
 
-    let greaterRarity: ArtifactRarity | undefined;
-
-    if (
-      fromActiveArtifact?.artifactType === ArtifactType.Wormhole &&
-      fromActiveArtifact.linkTo === toPlanet.locationId
-    ) {
-      greaterRarity = fromActiveArtifact.rarity;
-    }
-
-    if (
-      toActiveArtifact?.artifactType === ArtifactType.Wormhole &&
+    const fromHasActiveWormhole = fromActiveArtifact?.artifactType === ArtifactType.Wormhole &&
+      fromActiveArtifact.linkTo === toPlanet.locationId;
+    const toHasActiveWormhole = toActiveArtifact?.artifactType === ArtifactType.Wormhole &&
       toActiveArtifact.linkTo === fromPlanet.locationId
-    ) {
-      if (greaterRarity === undefined) {
+
+    let greaterRarity: ArtifactRarity | undefined = undefined;
+    switch(true) {
+      // active wormhole on both from and to planets choose the biggest rarity
+      case fromHasActiveWormhole && toHasActiveWormhole: {
+        greaterRarity = Math.max(fromActiveArtifact.rarity, toActiveArtifact.rarity) as ArtifactRarity;
+        break;
+      }
+      // only from planet has active wormhole, use that one
+      case fromHasActiveWormhole: {
+        greaterRarity = fromActiveArtifact.rarity;
+        break;
+      }
+      // only destination planet has active wormhole, use that one
+      case toHasActiveWormhole: {
```

Comment:
Using a pattern matching technique, which i think is more explicit about what we are setting the values to here.